// @ant-design/charts/lib/graphs/organizationalGraph/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var g6_1 = __importDefault(require("@antv/g6"));
var createLoading_1 = __importDefault(require("../../util/createLoading"));
var base_1 = require("../../base");
var useGraph_1 = __importDefault(require("../../hooks/useGraph"));
var customItems_1 = require("../customItems");
var constants_1 = require("../constants");
var utils_1 = require("../utils");
var defaultNodeStyle = {
    fill: '#91d5ff',
    stroke: '#40a9ff',
    radius: 2,
};
var defaultLayout = {
    type: 'compactBox',
    direction: 'TB',
    getId: function getId(d) {
        return d.id;
    },
    getHeight: function getHeight() {
        return 16;
    },
    getWidth: function getWidth() {
        return 16;
    },
    getVGap: function getVGap() {
        return 40;
    },
    getHGap: function getHGap() {
        return 70;
    },
};
var graphs = {};
var OrganizationalGraph = function (props) {
    var data = props.data, className = props.className, style = props.style, width = props.width, height = props.height, _a = props.animate, animate = _a === void 0 ? true : _a, _b = props.nodeType, nodeType = _b === void 0 ? 'rect' : _b, _c = props.edgeType, edgeType = _c === void 0 ? 'flow-line' : _c, _d = props.nodeSize, nodeSize = _d === void 0 ? constants_1.defaultNodeSize : _d, _e = props.behaviors, behaviors = _e === void 0 ? ['drag-canvas', 'zoom-canvas'] : _e, _f = props.nodeLabelCfg, nodeLabelCfg = _f === void 0 ? constants_1.defaultLabelCfg : _f, nodeCfg = props.nodeCfg, _g = props.layout, layout = _g === void 0 ? defaultLayout : _g, _h = props.showMarker, showMarker = _h === void 0 ? false : _h, _j = props.showArrow, showArrow = _j === void 0 ? true : _j, _k = props.arrowType, arrowType = _k === void 0 ? 'triangle' : _k, minimapCfg = props.minimapCfg, edgeCfg = props.edgeCfg, markerStyle = props.markerStyle, _l = props.nodeStateStyles, nodeStateStyles = _l === void 0 ? constants_1.defaultStateStyles : _l, _m = props.edgeStateStyles, edgeStateStyles = _m === void 0 ? constants_1.defaultStateStyles : _m, _o = props.autoFit, autoFit = _o === void 0 ? true : _o, onReady = props.onReady, loading = props.loading, loadingTemplate = props.loadingTemplate, errorTemplate = props.errorTemplate;
    var container = react_1.default.useRef(null);
    var graph = react_1.default.useRef(null);
    var graphId = utils_1.getGraphId(graph);
    useGraph_1.default(graphs[graphId], props, container);
    var arrowOffset = (Array.isArray(nodeSize) ? nodeSize[1] : nodeSize) / 2;
    react_1.useEffect(function () {
        var graphSize = utils_1.getGraphSize(width, height, container);
        if (nodeType === 'icon-node') {
            customItems_1.registerIconNode();
        }
        var graph = graphs[graphId];
        if (!graph) {
            graph = new g6_1.default.TreeGraph({
                container: container.current,
                width: graphSize[0],
                height: graphSize[1],
                linkCenter: true,
                animate: animate,
                modes: {
                    default: behaviors,
                },
                defaultNode: {
                    type: nodeType,
                    size: nodeSize,
                    labelCfg: nodeLabelCfg,
                    markerStyle: markerStyle,
                    showMarker: showMarker,
                },
                defaultEdge: {
                    type: edgeType,
                },
                nodeStateStyles: nodeStateStyles,
                edgeStateStyles: edgeStateStyles,
                layout: __assign(__assign({}, defaultLayout), layout),
            });
            graphs[graphId] = graph;
        }
        graph.node(function (node) {
            if (typeof nodeCfg === 'function') {
                return nodeCfg(node, graph);
            }
            return {
                style: __assign(__assign({}, defaultNodeStyle), nodeCfg === null || nodeCfg === void 0 ? void 0 : nodeCfg.style),
            };
        });
        graph.edge(function (edge) {
            if (typeof edgeCfg === 'function') {
                return edgeCfg(edge, graph);
            }
            return __assign(__assign({}, edgeCfg), { style: __assign(__assign({ stroke: '#91d5ff' }, (showArrow && utils_1.getDefaultEdgeArrowCfg(arrowOffset, arrowType, '#91d5ff'))), edgeCfg === null || edgeCfg === void 0 ? void 0 : edgeCfg.style) });
        });
        utils_1.processMinimap(minimapCfg, graph);
        utils_1.renderGraph(graph, data, autoFit);
        if (onReady) {
            onReady(graph);
        }
        return function () {
            if (graphs[graphId]) {
                graphs[graphId].destroy();
                delete graphs[graphId];
            }
        };
    }, []);
    return (react_1.default.createElement(base_1.ErrorBoundary, { errorTemplate: errorTemplate },
        loading && react_1.default.createElement(createLoading_1.default, { loadingTemplate: loadingTemplate }),
        react_1.default.createElement("div", { className: className, style: style, ref: container })));
};
exports.default = OrganizationalGraph;
