// @ant-design/charts/es/graphs/indentedTreeGraph/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React, { useEffect } from 'react';
import G6 from '@antv/g6';
import { ErrorBoundary } from '../../base';
import useGraph from '../../hooks/useGraph';
import ChartLoading from '../../util/createLoading';
import { getGraphSize, getGraphId } from '../utils';
import { bindDefaultEvents, renderGraph, getDefaultEdgeArrowCfg } from '../utils';
import { registerCardNode } from '../customItems';
import { defaultNodeAnchorPoints, defaultNodeSize, defaultStateStyles, defaultNodeStyle, } from '../constants';
var graphs = {};
registerCardNode();
var defaultLayout = {
    type: 'compactBox',
    direction: 'LR',
    getId: function (d) {
        return d.id;
    },
    getHeight: function () {
        return 60;
    },
    getWidth: function () {
        return 16;
    },
    getVGap: function () {
        return 16;
    },
    getHGap: function () {
        return 100;
    },
};
var IndentedTreeGraph = function (props) {
    var data = props.data, className = props.className, style = props.style, width = props.width, height = props.height, _a = props.nodeType, nodeType = _a === void 0 ? 'card' : _a, _b = props.edgeType, edgeType = _b === void 0 ? 'cubic-horizontal' : _b, _c = props.behaviors, behaviors = _c === void 0 ? ['zoom-canvas', 'drag-canvas'] : _c, _d = props.nodeAnchorPoints, nodeAnchorPoints = _d === void 0 ? defaultNodeAnchorPoints : _d, _e = props.nodeSize, nodeSize = _e === void 0 ? defaultNodeSize : _e, layout = props.layout, _f = props.animate, animate = _f === void 0 ? true : _f, nodeStyle = props.nodeStyle, edgeStyle = props.edgeStyle, edgeCfg = props.edgeCfg, markerStyle = props.markerStyle, _g = props.markerPosition, markerPosition = _g === void 0 ? 'right' : _g, _h = props.nodeStateStyles, nodeStateStyles = _h === void 0 ? defaultStateStyles : _h, _j = props.edgeStateStyles, edgeStateStyles = _j === void 0 ? defaultStateStyles : _j, _k = props.collapseExpand, collapseExpand = _k === void 0 ? true : _k, titleStyle = props.titleStyle, bodyStyle = props.bodyStyle, footerStyle = props.footerStyle, footerValueStyle = props.footerValueStyle, _l = props.showArrow, showArrow = _l === void 0 ? true : _l, _m = props.arrowType, arrowType = _m === void 0 ? 'vee' : _m, _o = props.autoFit, autoFit = _o === void 0 ? true : _o, onReady = props.onReady, loading = props.loading, loadingTemplate = props.loadingTemplate, errorTemplate = props.errorTemplate;
    var container = React.useRef(null);
    var graph = React.useRef(null);
    var graphId = getGraphId(graph);
    useGraph(graphs[graphId], props, container);
    useEffect(function () {
        var graphSize = getGraphSize(width, height, container);
        var graph = graphs[graphId];
        if (!graph) {
            graph = new G6.TreeGraph({
                container: container.current,
                width: graphSize[0],
                height: graphSize[1],
                animate: animate,
                modes: {
                    default: behaviors,
                },
                defaultNode: {
                    type: nodeType,
                    size: nodeSize,
                    anchorPoints: nodeAnchorPoints,
                    titleStyle: titleStyle,
                    bodyStyle: bodyStyle,
                    footerStyle: footerStyle,
                    footerValueStyle: footerValueStyle,
                    markerStyle: markerStyle,
                    collapseExpand: collapseExpand,
                    markerPosition: markerPosition,
                },
                defaultEdge: {
                    type: edgeType,
                },
                nodeStateStyles: nodeStateStyles,
                edgeStateStyles: edgeStateStyles,
                layout: __assign(__assign({}, defaultLayout), layout),
            });
            graphs[graphId] = graph;
        }
        graph.node(function (node) {
            if (typeof nodeStyle === 'function') {
                return {
                    style: nodeStyle(node, graph),
                };
            }
            return {
                style: __assign(__assign({}, defaultNodeStyle), nodeStyle),
            };
        });
        graph.edge(function (edge) {
            if (edgeCfg) {
                return typeof edgeCfg === 'function' ? edgeCfg(edge, graph) : edgeCfg;
            }
            if (typeof edgeStyle === 'function') {
                return {
                    style: edgeStyle(edge, graph),
                };
            }
            return {
                style: __assign(__assign({ stroke: '#ccc' }, (showArrow && getDefaultEdgeArrowCfg(0, arrowType))), edgeStyle),
            };
        });
        if (collapseExpand) {
            bindDefaultEvents(graph, collapseExpand);
        }
        renderGraph(graph, data, autoFit);
        if (onReady) {
            onReady(graph);
        }
        return function () {
            if (graphs[graphId]) {
                graphs[graphId].destroy();
                delete graphs[graphId];
            }
        };
    }, []);
    return (React.createElement(ErrorBoundary, { errorTemplate: errorTemplate },
        loading && React.createElement(ChartLoading, { loadingTemplate: loadingTemplate }),
        React.createElement("div", { className: className, style: style, ref: container })));
};
export default IndentedTreeGraph;
